homeassistant/components/directv/media_player.py

```diff
@@ -178,17 +157,32 @@ def __init__(self, name, device, dtv, version_info=None):
         self._is_client = device != "0"
         self._assumed_state = None
         self._available = False
+        self._enabled_default = enabled_default
         self._first_error_timestamp = None
+        self._model = None
+        self._receiver_id = None
+        self._software_version = None
 
-        if device != "0":
+        if self._is_client:
+            self._model = MODEL_CLIENT
             self._unique_id = device
-        elif version_info:
-            self._unique_id = "".join(version_info.get("receiverId").split())
+
+        if version_info:
+            self._receiver_id = "".join(version_info["receiverId"].split())
+
+            if not self._is_client:
+                self._unique_id = self._receiver_id
+                self._model = MODEL_HOST
+                self._software_version = version_info["stbSoftwareVersion"]
 
         if self._is_client:
-            _LOGGER.debug("Created DirecTV client %s for device %s", self._name, device)
+            _LOGGER.debug(
```

Comment:
We don't want side effects in init method. Please move this outside init or remove it.

Reply:
Thanks Martin. I plan to do further cleanup some of the logic and tests can prob be ripped out given the move to config flow but was trying to avoid further in this iteration

Reply:
https://github.com/home-assistant/core/pull/32809